src/components/DateTimePicker/DateTimePicker.js

```diff
@@ -76,81 +68,117 @@ const DateTimePicker = (props) => {
       onKeyDown();
     }
   };
+
   const handleTimeChange = (newTime) => {
     const dateWithOldTime = date ? new Date(date) : new Date();
     const hours = newTime ? newTime.getHours() : 0;
     const minutes = newTime ? newTime.getMinutes() : 0;
     const dateWithUpdatedTime = new Date(
-      dateWithOldTime.setHours(hours, minutes),
+      dateWithOldTime.setHours(hours, minutes, 0),
     );
     setDate(dateWithUpdatedTime);
   };
 
+  const clearTime = () => handleTimeChange(new Date(new Date().setHours(0, 0, 0)));
+
   const handleDateChange = (newDate) => {
-    setDate(newDate);
+    if (!newDate) {
+      return setDate(null);
+    }
+    const oldDate = date ? new Date(date) : new Date();
+    const hours = oldDate.getHours();
+    const minutes = oldDate.getMinutes();
+    const updatedDate = new Date(newDate.setHours(hours, minutes, 0));
+    return setDate(updatedDate);
   };
 
+  const clearDate = () => setDate(null);
+
   const handleDateTimeChange = (changedDate) => {
-    if (!hasDate) {
+    if (hasTime) {
       return handleTimeChange(changedDate);
     }
     return handleDateChange(changedDate);
   };
+
+  const handleHiddenInputChange = (e) => {
+    const { value } = e.target;
+    if (!value) {
+      return hasDate ? clearDate() : clearTime();
+    }
+    if (hasDate) {
+      return handleDateChange(new Date(value));
+    }
+    return handleTimeChange(
+      new Date(`${new Date().toDateString()} ${value}:00`),
```

Comment:
This seems very very scary :)

Reply:
And again one more SCARY:)

Reply:
scary stuff now hidden under date-fns parse method